packages/backend/src/secret.ts

```diff
@@ -27,6 +27,10 @@ import { BackendSecretFetcherFactory } from './engine/backend-secret/backend_sec
  *          ```
  */
 export const secret = (name: string): BackendSecret => {
+  // @ts-expect-error - this is a runtime check to prevent client-side usage
+  if (typeof window !== 'undefined' || typeof document !== 'undefined') {
+    throw new Error('Secrets are not supported in client-side applications');
```

Comment:
Regarding `client-side` . The CLI, IaC is kind of "client side". Should we just say `browser` ?

Reply:
that sounds fine, updated